gnutls/Interoperability/signature_algorithms-with-OpenSSL/runtest.sh

```diff
@@ -0,0 +1,345 @@
+            options+=(-CAfile $(x509Cert ca) -tls1_2)
+            options+=(-sigalgs ${S_OPENSSL[$idx]})
+            options+=(-client_sigalgs ${S_OPENSSL[$idx]})
+            options+=(-cert ${S_CLNT_CERT[$idx]} -key ${S_CLNT_KEY[$idx]})
+            rlRun "${options[*]} < /dev/null"
+            rlAssertGrep "Server Signature: ${S_NAME[$idx]}" server.log
+            rlAssertGrep "Client Signature: ${S_NAME[$idx]}" server.log
+
+            rlRun "kill $gnutls_pid" 0,1
+            rlRun "rlWait -s 9 $gnutls_pid" 143,1
+            if ! rlGetPhaseState; then
+                rlRun "cat server.log"
+                rlRun "cat server.err"
+            fi
+        rlPhaseEnd
+
+        rlPhaseStartTest "OpenSSL <-> GnuTLS [${S_NAME[$idx]}]"
+            # OpenSSL server setup
+            options=(openssl s_server -www -key ${S_KEY[$idx]})
+            options+=(-cert ${S_CERT[$idx]} -tls1_2)
+            options+=(-sigalgs ${S_OPENSSL[$idx]})
+            options+=(-CAfile '<(cat $(x509Cert ca) ${S_SUBCA[$idx]})')
+            rlRun "${options[*]} >server.log 2>server.err &"
+            openssl_pid=$!
+            rlRun "rlWaitForSocket 4433 -p $openssl_pid"
+
+            # GnuTLS client setup
+            options=(gnutls-cli -p 4433 --x509cafile $(x509Cert ca))
+            options+=(--priority ${GNUTLS_PROFILE}:+${S_GNUTLS[$idx]})
+            rlRun -s "${options[*]} localhost < /dev/null"
+            rlAssertGrep "Server Signature: ${S_NAME[$idx]}" "$rlRun_LOG"
+
+            rlRun "kill $openssl_pid" 0,1
+            rlRun "rlWait -s 9 $openssl_pid" 143,1
+            if ! rlGetPhaseState; then
+                rlRun "cat server.log"
+                rlRun "cat server.err"
+            fi
+        rlPhaseEnd
+
+        rlPhaseStartTest "OpenSSL <-> GnuTLS [${S_NAME[$idx]}, client auth]"
+            # OpenSSL server setup
+            options=(openssl s_server -www -key ${S_KEY[$idx]})
+            options+=(-cert ${S_CERT[$idx]} -tls1_2)
+            options+=(-sigalgs ${S_OPENSSL[$idx]})
+            options+=(-client_sigalgs ${S_OPENSSL[$idx]})
+            options+=(-CAfile '<(cat $(x509Cert ca) ${S_SUBCA[$idx]})')
+            options+=(-Verify 1 -verify_return_error)
+            rlRun "${options[*]} >server.log 2>server.err &"
+            openssl_pid=$!
+            rlRun "rlWaitForSocket 4433 -p $openssl_pid"
+
+            # GnuTLS client setup
+            options=(gnutls-cli -p 4433)
+            options+=(--x509cafile '<(cat $(x509Cert ca) ${C_SUBCA[$j]})')
+            options+=(--priority ${GNUTLS_PROFILE}:+${S_GNUTLS[$idx]})
+            options+=(--x509certfile ${S_CLNT_CERT[$idx]})
+            options+=(--x509keyfile ${S_CLNT_KEY[$idx]})
+
+            rlRun -s "${options[*]} localhost < /dev/null"
+            rlAssertGrep "Server Signature: ${S_NAME[$idx]}" "$rlRun_LOG"
+            rlAssertGrep "Client Signature: ${S_NAME[$idx]}" "$rlRun_LOG"
+
+            rlRun "kill $openssl_pid" 0,1
```

Comment:
-verify_return_error won't close/crash the server if the client certificate is invalid, but yes, it would be wise to check for potential segfault after a successful handshake